nomad/structs/secure_variables.go

```diff
@@ -28,6 +28,65 @@ type SecureVariableData struct {
 	KeyID string // ID of root key used to encrypt this entry
 }
 
+func (sv SecureVariableData) Copy() *SecureVariableData {
+	out := make([]byte, 0, len(sv.Data))
+	copy(out, sv.Data)
+	return &SecureVariableData{
+		Data:  out,
+		KeyID: sv.KeyID,
+	}
+}
+
+func (sv SecureVariable) Copy() SecureVariable {
+	out := SecureVariable{
+		Namespace:   sv.Namespace,
+		Path:        sv.Path,
+		CreateIndex: sv.CreateIndex,
+		CreateTime:  sv.CreateTime,
+		ModifyIndex: sv.ModifyIndex,
+		ModifyTime:  sv.ModifyTime,
+	}
```

Comment:
Primitive fields are copy-on-assignment. So typically we do this:

```suggestion
func (sv *SecureVariable) Copy() *SecureVariable {
	if sv == nil {
		return nil
	}
	out := *sv
```

It looks slightly "clever" but it avoids having to remember to add every new field to the `Copy` method except for the collections (maps, pointers, and slices).